c/include/cccl/types.h

```diff
@@ -0,0 +1,86 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of CUDA Experimental in CUDA C++ Core Libraries,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#ifndef CCCL_C_EXPERIMENTAL
+#  warning "C exposure is experimental and subject to change."
+#  warning "Define CCCL_C_EXPERIMENTAL to get it."
+#else
+
+#  if defined(_WIN32)
+#    define CCCL_C_API __declspec(dllexport)
```

Comment:
We do have the internal `_LIBCUDACXX_EXPORTED_FROM_ABI` macro for that.

I would love to get a new one one into the `__cccl/visibility.h` header That we can use everywhere

Reply:
It seems to be hidden under different macro. Could you please file an issue and productize it?

Reply:
I think I have a branch for that sitting around somewhere, because it is not trivial at all